opam-devel 2.0.0~rc fails to build

`opam install opam-devel -v` fails with

```
=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[opam-devel: ./configure] Command started
+ ./configure "--disable-checks" "--prefix" "/home/r/.opam/system" (CWD=/home/r/.opam/system/build/opam-devel.2.0.0~rc)
- checking for ocamlc... ocamlc
- OCaml version is 4.05.0
- OCaml library path is /usr/lib/ocaml
- checking for ocamlopt... ocamlopt
- checking for ocamlc.opt... ocamlc.opt
- checking for ocamlopt.opt... ocamlopt.opt
- checking for ocaml... ocaml
- checking for ocamldep... ocamldep
- checking for ocamldep.opt... ocamldep.opt
- checking for ocamlmktop... ocamlmktop
- checking for ocamlmklib... ocamlmklib
- checking for ocamldoc... ocamldoc
- checking for ocamldoc.opt... ocamldoc.opt
- checking for ocamlbuild... ocamlbuild
- checking OCaml Sys.os_type... Unix
- checking for gawk... gawk
- checking for compiler type... cc
- checking for compiler architecture... amd64
- checking for gcc... gcc -O2 -fno-strict-aliasing -fwrapv -D_FILE_OFFSET_BITS=64
- checking whether the C compiler works... yes
- checking for C compiler default output file name... a.out
- checking for suffix of executables... 
- checking whether we are cross compiling... no
- checking for suffix of object files... o
- checking whether we are using the GNU C compiler... yes
- checking whether gcc -O2 -fno-strict-aliasing -fwrapv -D_FILE_OFFSET_BITS=64 accepts -g... yes
- checking for gcc -O2 -fno-strict-aliasing -fwrapv -D_FILE_OFFSET_BITS=64 option to accept ISO C89... none needed
- checking for ocamlobjinfo... ocamlobjinfo
- checking for ocamlfind... ocamlfind
- checking for curl... curl
- checking for jbuilder... jbuilder
- 
- checking for OCaml findlib package unix... found
- checking for OCaml findlib package bigarray... found
- checking for OCaml findlib package extlib... found
- checking for OCaml findlib package re... found
- checking for OCaml findlib package re.glob... found
- checking for OCaml findlib package cmdliner... found
- checking for OCaml findlib package ocamlgraph... found
- checking for OCaml findlib package cudf... found
- checking for OCaml findlib package dose3.common... found
- checking for OCaml findlib package dose3.algo... found
- checking for OCaml findlib package opam-file-format... found
- checking for OCaml findlib package mccs... found
- checking for OCaml findlib package cppo... found
- 
- configure: creating ./config.status
- config.status: creating Makefile.config
- 
- Opam will be built WITH a built-in solver
- 
- Executables will be installed in /home/r/.opam/system/bin
- Manual pages will be installed in /home/r/.opam/system/share/man
[opam-devel: make opam-devel.install] Command started
+ make "opam-devel.install" (CWD=/home/r/.opam/system/build/opam-devel.2.0.0~rc)
- jbuilder build  -p opam opam.install
- Makefile:89: recipe for target 'opam-devel.install' failed
-        ocaml (internal)
- fatal: Not a git repository (or any parent up to mount point /)
- Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
- File "src/jbuild", line 6, characters 81-89:
- Warning: The use of the variable prefix '!' is deprecated, simply use '${^}'
- 
-         echo src/client/linking.sexp
-     ocamldep src/client/opamMain.depends.ocamldep-output
-        ocaml src/ocaml-flags-standard.sexp
-        ocaml src/ocaml-context-flags.sexp
-     ocamldep src/client/opamMain.dependsi.ocamldep-output
-       ocamlc src/client/opamMain.{cmi,cmti}
-     ocamlopt src/client/opamMain.{cmx,o}
-     ocamlopt src/client/opamMain.exe (exit 2)
- (cd _build/default && /usr/bin/ocamlopt.opt -w -40 -w +a-4-44-48 -safe-string -g -o src/client/opamMain.exe -I /home/r/.opam/system/lib/bytes -I /home/r/.opam/system/lib/cmdliner -I /home/r/.opam/system/lib/cudf -I /home/r/.opam/system/lib/dose3 -I /home/r/.opam/system/lib/extlib -I /home/r/.opam/system/lib/mccs -I /home/r/.opam/system/lib/mccs/glpk/internal -I /home/r/.opam/system/lib/ocamlgraph -I /home/r/.opam/system/lib/opam-client -I /home/r/.opam/system/lib/opam-core -I /home/r/.opam/system/lib/opam-file-format -I /home/r/.opam/system/lib/opam-format -I /home/r/.opam/system/lib/opam-repository -I /home/r/.opam/system/lib/opam-solver -I /home/r/.opam/system/lib/opam-state -I /home/r/.opam/system/lib/re -I /home/r/.opam/system/lib/result -I /usr/lib/ocaml /home/r/.opam/system/lib/result/result.cmxa /home/r/.opam/system/lib/extlib/extLib.cmxa /home/r/.opam/system/lib/re/re.cmxa /home/r/.opam/system/lib/re/re_perl.cmxa /home/r/.opam/system/lib/re/re_pcre.cmxa /home/r/.opam/system/lib/cudf/cudf.cmxa /home/r/.opam/system/lib/ocamlgraph/graph.cmxa /home/r/.opam/system/lib/dose3/common.cmxa /home/r/.opam/system/lib/mccs/glpk/internal/mccs_glpk.cmxa /usr/lib/ocaml/unix.cmxa /usr/lib/ocaml/bigarray.cmxa /home/r/.opam/system/lib/opam-core/opam_core.cmxa /home/r/.opam/system/lib/opam-file-format/opam-file-format.cmxa /home/r/.opam/system/lib/re/re_posix.cmxa /home/r/.opam/system/lib/dose3/algo.cmxa /home/r/.opam/system/lib/mccs/mccs.cmxa /home/r/.opam/system/lib/opam-format/opam_format.cmxa /home/r/.opam/system/lib/opam-repository/opam_repository.cmxa /home/r/.opam/system/lib/cmdliner/cmdliner.cmxa /home/r/.opam/system/lib/opam-solver/opam_solver.cmxa /home/r/.opam/system/lib/opam-state/opam_state.cmxa /home/r/.opam/system/lib/re/re_glob.cmxa /home/r/.opam/system/lib/opam-client/opam_client.cmxa src/client/opamMain.cmx)
- File "_none_", line 1:
- Error: No implementations provided for the following modules:
-          Unix referenced from /home/r/.opam/system/lib/dose3/common.cmxa(Common)
- make: *** [opam-devel.install] Error 1
[ERROR] The compilation of opam-devel failed at "make opam-devel.install".

#=== ERROR while installing opam-devel.2.0.0~rc ===============================#
# opam-version 1.2.2
# os           linux
# command      make opam-devel.install
# path         /home/r/.opam/system/build/opam-devel.2.0.0~rc
# compiler     system (4.05.0)
# exit-code    2
# env-file     /home/r/.opam/system/build/opam-devel.2.0.0~rc/opam-devel-15458-e7966f.env
# stdout-file  /home/r/.opam/system/build/opam-devel.2.0.0~rc/opam-devel-15458-e7966f.out
# stderr-file  /home/r/.opam/system/build/opam-devel.2.0.0~rc/opam-devel-15458-e7966f.err
### stdout ###
# jbuilder build  -p opam opam.install
# Makefile:89: recipe for target 'opam-devel.install' failed
### stderr ###
# Warning: The use of the variable prefix '!' is deprecated, simply use '${^}'
# [...]
#     ocamldep src/client/opamMain.dependsi.ocamldep-output
#       ocamlc src/client/opamMain.{cmi,cmti}
#     ocamlopt src/client/opamMain.{cmx,o}
#     ocamlopt src/client/opamMain.exe (exit 2)
# (cd _build/default && /usr/bin/ocamlopt.opt -w -40 -w +a-4-44-48 -safe-string -g -o src/client/opamMain.exe -I /home/r/.opam/system/lib/bytes -I /home/r/.opam/system/lib/cmdliner -I /home/r/.opam/system/lib/cudf -I /home/r/.opam/system/lib/dose3 -I /home/r/.opam/system/lib/extlib -I /home/r/.opam/system/lib/mccs -I /home/r/.opam/system/lib/mccs/glpk/internal -I /home/r/.opam/system/lib/ocamlgraph -I /home/r/.opam/system/lib/opam-client -I /home/r/.opam/system/lib/opam-core -I /home/r/.opam/system/lib/opam-file-format -I /home/r/.opam/system/lib/opam-format -I /home/r/.opam/system/lib/opam-repository -I /home/r/.opam/system/lib/opam-solver -I /home/r/.opam/system/lib/opam-state -I /home/r/.opam/system/lib/re -I /home/r/.opam/system/lib/result -I /usr/lib/ocaml /home/r/.opam/system/lib/result/result.cmxa /home/r/.opam/system/lib/extlib/extLib.cmxa /home/r/.opam/system/lib/re/re.cmxa /home/r/.opam/system/lib/re/re_perl.cmxa /home/r/.opam/system/lib/re/re_pcre.cmxa /home/r/.opam/system/lib/cudf/cudf.cmxa /home/r/.opam/system/lib/ocamlgraph/graph.cmxa /home/r/.opam/system/lib/dose3/common.cmxa /home/r/.opam/system/lib/mccs/glpk/internal/mccs_glpk.cmxa /usr/lib/ocaml/unix.cmxa /usr/lib/ocaml/bigarray.cmxa /home/r/.opam/system/lib/opam-core/opam_core.cmxa /home/r/.opam/system/lib/opam-file-format/opam-file-format.cmxa /home/r/.opam/system/lib/re/re_posix.cmxa /home/r/.opam/system/lib/dose3/algo.cmxa /home/r/.opam/system/lib/mccs/mccs.cmxa /home/r/.opam/system/lib/opam-format/opam_format.cmxa /home/r/.opam/system/lib/opam-repository/opam_repository.cmxa /home/r/.opam/system/lib/cmdliner/cmdliner.cmxa /home/r/.opam/system/lib/opam-solver/opam_solver.cmxa /home/r/.opam/system/lib/opam-state/opam_state.cmxa /home/r/.opam/system/lib/re/re_glob.cmxa /home/r/.opam/system/lib/opam-client/opam_client.cmxa src/client/opamMain.cmx)
# File "_none_", line 1:
# Error: No implementations provided for the following modules:
#          Unix referenced from /home/r/.opam/system/lib/dose3/common.cmxa(Common)
# make: *** [opam-devel.install] Error 1
```

This is using opam 1.2.2 and a system switch with OCaml 4.05.

--- a/packages/dose3/dose3.4.3/files/0004-Add-unix-as-dependency-to-dose3.common-in-META.in.patch
+++ b/packages/dose3/dose3.4.3/files/0004-Add-unix-as-dependency-to-dose3.common-in-META.in.patch
@@ -1,0 +1,25 @@
+From b94cf24739818e5aff397e0a83b19ea32dc81f42 Mon Sep 17 00:00:00 2001
+From: Louis Gesbert <louis.gesbert@ocamlpro.com>
+Date: Tue, 6 Feb 2018 10:15:45 +0100
+Subject: [PATCH 3/3] Add "unix" as dependency to dose3.common in META.in
+
+---
+ META.in | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/META.in b/META.in
+index aa2cd8d..0f9d337 100644
+--- a/META.in
++++ b/META.in
+@@ -8,7 +8,7 @@ package "common" (
+ version = "@PACKAGE_VERSION@"
+ archive(byte) = "common.cma"
+ archive(native) = "common.cmxa"
+-requires = "extlib, re.pcre, cudf, @ZIP@, @BZ2@"
++requires = "extlib, re.pcre, cudf, unix, @ZIP@, @BZ2@"
+ )
+ 
+ package "algo" (
+-- 
+2.11.0
+

--- a/packages/dose3/dose3.4.3/opam
+++ b/packages/dose3/dose3.4.3/opam
@@ -33,3 +33,6 @@ depends: [
   "cppo" {build & >= "1.1.2"}
 ]
 conflicts: "dose"
+patches: [
+  "0004-Add-unix-as-dependency-to-dose3.common-in-META.in.patch"
+]

--- a/packages/dose3/dose3.5.0/files/0004-Add-unix-as-dependency-to-dose3.common-in-META.in.patch
+++ b/packages/dose3/dose3.5.0/files/0004-Add-unix-as-dependency-to-dose3.common-in-META.in.patch
@@ -1,0 +1,25 @@
+From b94cf24739818e5aff397e0a83b19ea32dc81f42 Mon Sep 17 00:00:00 2001
+From: Louis Gesbert <louis.gesbert@ocamlpro.com>
+Date: Tue, 6 Feb 2018 10:15:45 +0100
+Subject: [PATCH 3/3] Add "unix" as dependency to dose3.common in META.in
+
+---
+ META.in | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/META.in b/META.in
+index aa2cd8d..0f9d337 100644
+--- a/META.in
++++ b/META.in
+@@ -8,7 +8,7 @@ package "common" (
+ version = "@PACKAGE_VERSION@"
+ archive(byte) = "common.cma"
+ archive(native) = "common.cmxa"
+-requires = "extlib, re.pcre, cudf, @ZIP@, @BZ2@"
++requires = "extlib, re.pcre, cudf, unix, @ZIP@, @BZ2@"
+ )
+ 
+ package "algo" (
+-- 
+2.11.0
+

--- a/packages/dose3/dose3.5.0/opam
+++ b/packages/dose3/dose3.5.0/opam
@@ -33,3 +33,6 @@ depends: [
   "cppo" {build & >= "1.1.2"}
 ]
 conflicts: "dose"
+patches: [
+  "0004-Add-unix-as-dependency-to-dose3.common-in-META.in.patch"
+]

